smmregrid/gridtype.py

```diff
@@ -0,0 +1,214 @@
+# GridType class to gather all information about grids with shared dimensions
+import xarray as xr
+
+# default spatial dimensions and vertical coordinates
+DEFAULT_DIMS = {
```

Comment:
What would happen if the dataset contains some dim with a name in this list but that dim is actually used for something else, not as a spatial coordinate? E.g 'i' could be an index on something else. 

The user can extend them with `extra_dims` but cannot substitute them, correct?

Reply:
MAybe would it be better to expose to the user the list of defaults, so that if the want they can make a new list + other names (extension' or they replace it entirely. In any case instead of extra dimensions a full alternative set of dimensions is passed ...